packages/workspace/src/utils/default-base.ts

```diff
@@ -0,0 +1,19 @@
+import { execSync } from 'child_process';
+
+export function deduceDefaultBase(fallback: string = 'main'): string {
+  let res: string;
+  try {
+    res = execSync('git config --get init.defaultBranch').toString().trim();
+  } catch {}
+  // If command fails or returns empty string, we set it to main as default.
+  return res || fallback;
+}
+
+export function checkGitVersion(): string | null {
+  try {
+    let gitVersionOutput = execSync('git --version').toString().trim();
+    return gitVersionOutput.match(/([0-9]+\.[0-9]+\.+[0-9]+)/)[0];
```

Comment:
It doesn't look like you're using the capture group :thinking:

```suggestion
    return gitVersionOutput.match(/[0-9]+\.[0-9]+\.+[0-9]+/)[0];
```